src/test/java/ca/uhn/fhir/jpa/starter/OperationCollectDataTest.java

```diff
@@ -0,0 +1,200 @@
+    assertTrue(bundle.getEntry().get(1).getResource() instanceof Observation);
+  }
+
+  @BeforeEach
+  void beforeEach() {
+    ourServerBase = "http://localhost:" + port + "/fhir";
+    setHapiClient();
+    ourHttpClient = HttpClientBuilder.create().build();
+  }
+
+  private Parameters fetchParameter(String theUrl) throws IOException, ClientProtocolException {
+    Parameters parameters;
+    HttpGet get = new HttpGet(theUrl);
+
+    String username = "hapi";
+    String password = "hapi123";
+
+    String auth = username + ":" + password;
+    byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+    String authHeader = "Basic " + new String(encodedAuth);
+    get.addHeader(HttpHeaders.AUTHORIZATION, authHeader);
+
+    get.addHeader(Constants.HEADER_CACHE_CONTROL, Constants.CACHE_CONTROL_NO_CACHE);
+
+    CloseableHttpResponse resp = ourHttpClient.execute(get);
+    try {
+      assertEquals(EncodingEnum.JSON.getResourceContentTypeNonLegacy(),
+          resp.getFirstHeader(ca.uhn.fhir.rest.api.Constants.HEADER_CONTENT_TYPE).getValue().replaceAll(";.*", ""));
+      parameters = EncodingEnum.JSON.newParser(ourCtx).parseResource(Parameters.class,
+          IOUtils.toString(resp.getEntity().getContent(), Charsets.UTF_8));
+    } finally {
+      IOUtils.closeQuietly(resp);
+    }
+    return parameters;
+  }
+
+  private CloseableHttpResponse postResource(String theUrl, String filePath)
+      throws IOException, ClientProtocolException {
+    HttpPost post = new HttpPost(theUrl);
+    String username = "hapi";
+    String password = "hapi123";
+    String json = readJsonFile(filePath);
+
+    StringEntity entity = new StringEntity(json);
+    post.setEntity(entity);
+    post.setHeader("Accept", "application/json");
+    post.setHeader("Content-type", "application/json");
+
+    String auth = username + ":" + password;
+    byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+    String authHeader = "Basic " + new String(encodedAuth);
+    post.addHeader(HttpHeaders.AUTHORIZATION, authHeader);
+
+    post.addHeader(Constants.HEADER_CACHE_CONTROL, Constants.CACHE_CONTROL_NO_CACHE);
+    CloseableHttpResponse resp = ourHttpClient.execute(post);
+    return resp;
+  }
+
+  private String readJsonFile(String path) throws IOException {
+    String content = null;
+    try {
+      content = FileUtils.readFileToString(new File(path), StandardCharsets.UTF_8);
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
+    }
+    return content;
+  }
+
+  private void setHapiClient() {
+    ourCtx = FhirContext.forR4();
+    ourCtx.getRestfulClientFactory().setServerValidationMode(ServerValidationModeEnum.NEVER);
+    ourCtx.getRestfulClientFactory().setSocketTimeout(1200 * 1000);
+    ourCtx.setParserErrorHandler(new StrictErrorHandler());
+
+    ourClient = ourCtx.newRestfulGenericClient(ourServerBase);
```

Comment:
if newRestfulGenericClient is expecting a live server, then we need to mock this out, if available hapi fhir client tests don't normally do this, mocking is the right way forward

Reply:
@kaweesi  , 
1 , i simply followed the convetions of the rest of the Tests , they invoke a live server , not Mocking.
2. I wanted to Test the actual behaviour of the server and the API , not just Mocks